src/gurt/telemetry.c

```diff
@@ -1894,10 +1903,11 @@ d_tm_init_histogram(struct d_tm_node_t *node, char *path, int num_buckets,
 	struct d_tm_bucket_t	*dth_buckets;
 	uint64_t		min = 0;
 	uint64_t		max = 0;
+	uint64_t		prev_width = 0;
```

Comment:
just wondering why you need this extra variable? can't you just use "max = min + ((i+1) * initial_width * multiplier) - 1" or doesn't that work?

Reply:
The formula you suggested did not give the bucket ranges that I thought we were going for.  With your formula, we get:
histogram bucket 0 [0 .. 2047] width: 2048
histogram bucket 1 [2048 .. 6143] width: 4096
histogram bucket 2 [6144 .. 14335] width: 8192
histogram bucket 3 [14336 .. 26623] width: 12288
histogram bucket 4 [26624 .. 43007] width: 16384
histogram bucket 5 [43008 .. 63487] width: 20480
histogram bucket 6 [63488 .. 88063] width: 24576
histogram bucket 7 [88064 .. 116735] width: 28672
histogram bucket 8 [116736 .. 149503] width: 32768
histogram bucket 9 [149504 .. 18446744073709551615] width: 18446744073709402112

It shows that the buckets go up in size linearly by 4K, because yours is based on the initial width * multiplier.

My formula uses the width of the previous bucket, so we get this progression:
histogram bucket 0 [0 .. 2047] width: 2048
histogram bucket 1 [2048 .. 6143] width: 4096
histogram bucket 2 [6144 .. 14335] width: 8192
histogram bucket 3 [14336 .. 30719] width: 16384
histogram bucket 4 [30720 .. 63487] width: 32768
histogram bucket 5 [63488 .. 129023] width: 65536
histogram bucket 6 [129024 .. 260095] width: 131072
histogram bucket 7 [260096 .. 522239] width: 262144
histogram bucket 8 [522240 .. 1046527] width: 524288
histogram bucket 9 [1046528 .. 18446744073709551615] width: 18446744073708505088

Which shows a doubling in width for each bucket.  I think the doubling in width is what we are after.  Do you disagree?

Reply:
no you are correct, thanks Joel